qiskit/synthesis/two_qubit/two_qubit_decompose.py

```diff
@@ -476,129 +240,27 @@ def __repr__(self):
 
     @classmethod
     def from_bytes(
-        cls, bytes_in: bytes, *, requested_fidelity: float, **kwargs
+        cls,
+        bytes_in: bytes,
+        *,
+        requested_fidelity: float,
+        _specialization: two_qubit_decompose.Specialization | None,
+        **kwargs,
     ) -> "TwoQubitWeylDecomposition":
         """Decode bytes into :class:`.TwoQubitWeylDecomposition`."""
         # Used by __repr__
         del kwargs  # Unused (just for display)
         b64 = base64.decodebytes(bytes_in)
         with io.BytesIO(b64) as f:
             arr = np.load(f, allow_pickle=False)
-        return cls(arr, fidelity=requested_fidelity)
+        return cls(arr, fidelity=requested_fidelity, _specialization=_specialization)
 
     def __str__(self):
         pre = f"{self.__class__.__name__}(\n\t"
```

Comment:
Now that this PR does away with the subclassing - this str() representation no longer directly tells which  specialization was applied (there are still clues from the form of the circuit chosen based on `euler_basis` and `weyl_gate`). Maybe would be nice to add an indication of the specialization back?

Reply:
Added it to `__str__` in: https://github.com/Qiskit/qiskit/pull/11946/commits/c52bffa7c071da01d1a9a21e0f15c73748617e84